mlflow/resource_webhook.go

```
@@ -0,0 +1,155 @@
+}
+
+type webhookListResponse struct {
+	Webhooks []Webhook `json:"webhooks"`
+}
+
+type webhookApiResponse struct {
+	Webhook Webhook `json:"webhook"`
+}
+
+type WebhooksAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewWebhooksAPI(ctx context.Context, m interface{}) WebhooksAPI {
+	return WebhooksAPI{m.(*common.DatabricksClient), ctx}
+}
+
+func (a WebhooksAPI) Create(m Webhook) (string, error) {
+	var r webhookApiResponse
+	err := a.client.Post(a.context, "/mlflow/registry-webhooks/create", &m, &r)
+	return r.Webhook.ID, err
+}
+
+func (a WebhooksAPI) Read(ID string) (Webhook, error) {
+	var m webhookListResponse
+	err := a.client.Get(a.context, "/mlflow/registry-webhooks/list", nil, &m)
+	if err != nil {
+		return Webhook{}, err
+	}
+	for _, wh := range m.Webhooks {
+		if wh.ID == ID {
+			return wh, nil
+		}
+	}
+	return Webhook{}, fmt.Errorf("Webhook with ID %s isn't found", ID)
+}
+
+// Update the webhook entity
+func (a WebhooksAPI) Update(id string, m Webhook) error {
+	// Update API doesn't allow to change Model name field after creation.
+	m.ModelName = ""
+	m.ID = id
+	return a.client.Patch(a.context, "/mlflow/registry-webhooks/update", m)
+}
+
+// Delete removes the webhook by its ID
+func (a WebhooksAPI) Delete(ID string) error {
+	return a.client.Delete(a.context, "/mlflow/registry-webhooks/delete", map[string]string{
+		"id": ID,
+	})
+}
+
+func ResourceMLFlowWebhook() *schema.Resource {
+	s := common.StructToSchema(
+		Webhook{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["status"].ValidateFunc = validation.StringInSlice([]string{"ACTIVE", "TEST_MODE", "DISABLED"}, true)
+			// TODO: do we need a validation for Events?
+			delete(m, "id")
+			p, err := common.SchemaPath(m, "http_url_spec", "enable_ssl_verification")
+			if err == nil {
+				p.Required = false
+				p.Optional = true
+			} else {
+				log.Printf("[WARN] Can't find http_url_spec.enable_ssl_verification in schema")
```

this statement is probably never reachable... ignore err as _ then